.github/workflows/build-test-distribute.yaml

```diff
@@ -0,0 +1,240 @@
+name: "build-test-distribute"
+on:
+  push:
+    branches: ["master", "release-*"]
+  pull_request_target:
+    branches: ["master", "release-*"]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+      # GitHub actions does not share cache across multiple jobs,
+      # so we have to operate cache in each job and action file
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.kuma-dev
+          key: ${{ runner.os }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-devtools
+      - run: |
+          make dev/tools
+      - run: |
+          make clean
+      - run: |
+          make check
+      - run: |
+          make build
+      - run: |
+          make -j build/distributions
+      - run: |
+          make -j images
+      - run: |
+          make -j docker/save
+      - name: Pack artifact
+        run: |
+          tar -czf build.tar.gz ./build
```

Comment:
The `executable` permission will be lost after executing "upload-artifact" and "download-artifact" actions, so make a tar before uploading. This takes ~1.5min.